src/api/procedures/modifyAllowance.ts

```diff
@@ -0,0 +1,145 @@
+  /**
+   * amount of POLYX to be increased
+   */
+  allowance: BigNumber;
+  operation: AllowanceOperation.Increase;
+}
+
+interface DecreaseAllowanceParams {
+  /**
+   * amount of POLYX to be decreased
+   */
+  allowance: BigNumber;
+  operation: AllowanceOperation.Decrease;
+}
+
+interface SetAllowanceParams {
+  /**
+   * amount of POLYX to be set
+   */
+  allowance: BigNumber;
+  operation: AllowanceOperation.Set;
+}
+
+export type ModifyAllowanceParams = (
+  | IncreaseAllowanceParams
+  | DecreaseAllowanceParams
+  | SetAllowanceParams
+) & {
+  subsidy: Subsidy;
+};
+
+/**
+ * @hidden
+ */
+export async function prepareModifyAllowance(
+  this: Procedure<ModifyAllowanceParams, void>,
+  args: ModifyAllowanceParams
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: { tx },
+    },
+    context,
+  } = this;
+
+  const {
+    subsidy: {
+      beneficiary: { address: beneficiaryAddress },
+    },
+    subsidy,
+    allowance,
+    operation,
+  } = args;
+
+  const [exists, currentAllowance] = await Promise.all([subsidy.exists(), subsidy.getAllowance()]);
+
+  if (!exists) {
+    throw new PolymeshError({
+      code: ErrorCode.DataUnavailable,
+      message: 'The Subsidy no longer exists',
+    });
+  }
+
+  const rawAllowance = bigNumberToBalance(allowance, context);
+
+  const rawBeneficiaryAccount = stringToAccountId(beneficiaryAddress, context);
+
+  let transaction = tx.relayer.increasePolyxLimit;
+
+  if (operation === AllowanceOperation.Set) {
+    if (currentAllowance.eq(allowance)) {
+      throw new PolymeshError({
+        code: ErrorCode.NoDataChange,
+        message: 'Amount of allowance to set is equal to the current allowance',
+      });
+    }
+
+    transaction = tx.relayer.updatePolyxLimit;
+  }
+
+  if (operation === AllowanceOperation.Decrease) {
+    if (currentAllowance.lte(allowance)) {
+      throw new PolymeshError({
+        code: ErrorCode.NoDataChange,
+        message: 'Amount of allowance to decrease cannot be more than the current allowance',
+      });
+    }
+
+    transaction = tx.relayer.decreasePolyxLimit;
+  }
+
+  this.addTransaction({
+    transaction,
+    args: [rawBeneficiaryAccount, rawAllowance],
+  });
+}
+
+/**
+ * @hidden
+ *
+ * To modify the allowance for a Subsidy, the caller should be the subsidizer
+ */
+export function getAuthorization(
+  this: Procedure<ModifyAllowanceParams, void>,
+  args: ModifyAllowanceParams
+): ProcedureAuthorization {
+  const { context } = this;
+  const {
+    subsidy: { subsidizer },
+    operation,
+  } = args;
+
+  const currentAccount = context.getCurrentAccount();
+
+  const hasRoles = subsidizer.isEqual(currentAccount);
+
+  const transactionMap = new Map<AllowanceOperation, RelayerTx>([
+    [AllowanceOperation.Increase, TxTags.relayer.IncreasePolyxLimit],
+    [AllowanceOperation.Decrease, TxTags.relayer.DecreasePolyxLimit],
+    [AllowanceOperation.Set, TxTags.relayer.UpdatePolyxLimit],
+  ]);
```

Comment:
```suggestion
    const transactionMap = {
    [AllowanceOperation.Increase]: TxTags.relayer.IncreasePolyxLimit,
    [AllowanceOperation.Decrease]: TxTags.relayer.DecreasePolyxLimit,
    [AllowanceOperation.Set]: TxTags.relayer.UpdatePolyxLimit,
  };
```

With an object TS will be able to tell it matches all cases and you won't need the no null assertion